Why: Do not build tools
Why: Remove GRegex pre-emptively
Why: Add support for WebAssembly
Why: Revert "Remove GRegex pre-emptively"
Source: Ported by Kleis Auke Wolthuizen (wasm-vips): https://github.com/GNOME/glib/compare/2.75.0...kleisauke:wasm-vips-2.75.0

diff --git a/glib/meson.build b/glib/meson.build
index 4130b1ba6f134b80c69bfe9fce3dd87ae31a2878..f9ece815f9d958d8fcce1e047344141fd9d931c2 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -362,9 +362,10 @@ if host_system == 'windows'
   if cc.get_id() == 'msvc' or cc.get_id() == 'clang-cl'
     glib_sources += files('dirent/wdirent.c')
   endif
+elif host_system == 'emscripten'
+  glib_sources += files('gspawn-wasm.c')
 else
   glib_sources += files('glib-unix.c', 'gspawn.c', 'giounix.c')
-  platform_deps = []
 endif
 
 if host_system == 'linux'
@@ -475,7 +476,7 @@ if host_system == 'windows'
       include_directories : configinc,
       dependencies : [libglib_dep])
   endif
-else
+elif host_system != 'emscripten'
   gtester = executable('gtester', 'gtester.c',
     install : true,
     install_tag : 'bin-devel',
